pkg/disruptors/pod.go

```diff
@@ -120,6 +130,37 @@ func (d *podDisruptor) InjectGrpcFaults(
 	options GrpcDisruptionOptions,
 ) error {
 	cmd := buildGrpcFaultCmd(fault, duration, options)
-	err := d.controller.ExecCommand(ctx, cmd)
+
+	err := d.portValidator.ValidatePort(ctx, fault.Port)
+	if err != nil {
+		return fmt.Errorf("verify fault port: %w", err)
+	}
+
+	err = d.controller.ExecCommand(ctx, cmd)
 	return err
 }
+
+// PortValidator is responsible for port validation of selected targets
+type PortValidator struct {
+	filter helpers.PodFilter
+	helper helpers.PodHelper
+}
+
+// NewPortValidator creates a new instance of PortValidator that acts on the pods that matches
+// the given pod filter
+func NewPortValidator(podFilter helpers.PodFilter, podHelper helpers.PodHelper) *PortValidator {
+	return &PortValidator{
+		filter: podFilter,
+		helper: podHelper,
+	}
+}
+
+// ValidatePort verifies if the given port complies with internal rules.
+// If 0 is provided as port, DefaultTargetPort is used instead
+func (pv *PortValidator) ValidatePort(ctx context.Context, port uint) error {
+	if port == 0 {
+		port = DefaultTargetPort
+	}
+
+	return pv.helper.ValidatePort(ctx, pv.filter, port)
```

Comment:
The PortValidator is basically just calling the helper function. Given that the pod disruptor already uses this helper I think we don't really need the validator. If I recall correctly, we introduced it just to facilitate testing. But as the validation logic is now a function of the Helper, we can test it there.

Reply:
Agree, for some reason I decided to avoid direct dependencies in PodDisruptor, but now I see it doesn't make sense